templates/env.rc

```diff
@@ -0,0 +1,122 @@
+  CLOUDS_PATH="$1"
+else
+  echo "Error: No clouds.yaml provided"
+  echo "You must provide a valid clouds.yaml script to generate a cloud.conf"
+  echo ""
+  exit 1
+fi
+
+# Check if os cloud is provided
+if [[ -n "${2-}" ]] && [[ $2 != -* ]] && [[ $2 != --* ]]; then
+  export CLOUD=$2
+else
+  echo "Error: No cloud specified"
+  echo "You must specify which cloud you want to use."
+  echo ""
+  exit 1
+fi
+
+yq_type=$(file "$(which yq)")
+if [[ ${yq_type} == *"Python script"* ]]; then
+  echo "Wrong version of 'yq' installed, please install the one from https://github.com/mikefarah/yq"
+  echo ""
+  exit 1
+fi
+
+CLOUDS_PATH=${CLOUDS_PATH:-""}
+OPENSTACK_CLOUD_YAML_CONTENT=$(cat "$CLOUDS_PATH")
+
+# Just blindly parse the cloud.yaml here, overwriting old vars.
+AUTH_URL=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.auth_url)
+USERNAME=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.username)
+PASSWORD=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.password)
+REGION=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.region_name)
+PROJECT_ID=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.project_id)
+PROJECT_NAME=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.project_name)
+DOMAIN_NAME=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.user_domain_name)
+if [[ "$DOMAIN_NAME" = "null" ]]; then
+  DOMAIN_NAME=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.domain_name)
+fi
+DOMAIN_ID=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.user_domain_id)
+if [[ "$DOMAIN_ID" = "null" ]]; then
+  DOMAIN_ID=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.auth.domain_id)
+fi
+CACERT_ORIGINAL=$(echo "$OPENSTACK_CLOUD_YAML_CONTENT" | yq r - clouds.${CLOUD}.cacert)
+
+# Build OPENSTACK_CLOUD_YAML_B64
+OPENSTACK_CLOUD_YAML_SELECTED_CLOUD_B64=$(echo "${OPENSTACK_CLOUD_YAML_CONTENT}" | yq r - clouds.${CLOUD} | yq p - clouds.${CLOUD} | base64 --wrap=0)
+export OPENSTACK_CLOUD_YAML_B64="${OPENSTACK_CLOUD_YAML_SELECTED_CLOUD_B64}"
+
+# Build OPENSTACK_CLOUD_PROVIDER_CONF_B64
+# Basic cloud.conf, no LB configuration as that data is not known yet.
+cloud_provider_conf_tmp=$(mktemp /tmp/cloud.confXXX)
+OPENSTACK_CLOUD_PROVIDER_CONF=""
+cat >> ${cloud_provider_conf_tmp} << EOF
+[Global]
+auth-url=$AUTH_URL
+username="$USERNAME"
+password="$PASSWORD"
+EOF
+
+if [[ "$PROJECT_ID" != "null" ]]; then
+  echo "tenant-id=\"${PROJECT_ID}\"" >> ${cloud_provider_conf_tmp}
+fi
+if [[ "$PROJECT_NAME" != "null" ]]; then
+  echo "tenant-name=\"${PROJECT_NAME}\"" >> ${cloud_provider_conf_tmp}
+fi
+if [[ "$DOMAIN_NAME" != "null" ]]; then
+  echo "domain-name=\"${DOMAIN_NAME}\"" >> ${cloud_provider_conf_tmp}
+fi
+if [[ "$DOMAIN_ID" != "null" ]]; then
+  echo "domain-id=\"${DOMAIN_ID}\"" >> ${cloud_provider_conf_tmp}
+fi
+
+if [[ "$CACERT_ORIGINAL" != "null" ]]; then
```

Comment:
I'm comparing against null because yq returns null if the key is not found in the YAML. Are you using: `GO111MODULE=on go get github.com/mikefarah/yq/v2` or the python yq?

Reply:
I am sorry I used snap yq. I confirmed this script works with `GO111MODULE=on go get github.com/mikefarah/yq/v2' , thanks.